### ISSUE TITLE
FNX-5541 ⁃ Clear Browsing Data

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/fenix#225]
As a user, I want to erase any traces of my history, or any saved cookie, passwords so nobody can see what I've done previously in the browser or any publishers following my trace

UX: #980 

### Acceptance Criteria
-I can go to settings to delete all browsing data 
-I see a toast to indicate that browsing data has been deleted

**Comment by bbinto** [comment]
As a user, I want to erase any traces of my history, or any saved cookie, passwords so nobody can see what I've done previously in the browser or any publishers following my trace

Be able to clear individual areas, and individual items within (e.g. clear 1 specific cookie, while form cache won't be deleted) 
- clear: all
- clear: on exit
- No cache

**Comment by davismtl** [comment]
For clearing browsing history, do you expect the deletions to sync to other devices?

**Comment by st3fan** [comment]
@bbinto What does "on exit" mean on mobile? Android apps have no "Quit" option. You never "exit" an Android app.

**Comment by st3fan** [comment]
There is a whole discussion with bugs and screenshots about this at https://github.com/mozilla-mobile/reference-browser/issues/81

**Comment by AmyYLee** [comment]
@bbinto This looks like a duplicate of this bug. I posted specs there.
https://github.com/mozilla-mobile/fenix/issues/980

**Comment by bbinto** [comment]
This issue is referring to the Eng tasks, so I think this should still be kept open (cc @vesta0)

**Comment by mheubusch** [comment]
Issue 980 is reviewed and signed off on by UX so this is ready to build.

**Comment by bifleming** [comment]
@vesta0 This story might be blocked by both A/C and GV.

**Comment by vesta0** [comment]
@cpeterso I understand that [this API ](https://bugzilla.mozilla.org/show_bug.cgi?id=1489669)is required to delete browsing data selectively, but do we also need a new GV API for a catch-all method that deletes everything?

**Comment by cpeterso** [comment]
The GV team implement a GV API (bug 1489669) to selectively clear user data by site (or for all sites) or by data type (or for all data types). Fenix MVP doesn't necessarily need to expose the by site or by data in the UI. Fenix can use this selective API to implement the simpler "clear all user data" functionality.

**Comment by cpeterso** [comment]
GV's Clear User Data API landed last week:
https://bugzilla.mozilla.org/show_bug.cgi?id=1489669#c22

And should be available in yesterday's GV 68 Beta build:
https://maven.mozilla.org/?prefix=maven2/org/mozilla/geckoview/geckoview-beta-armeabi-v7a/68.0.20190527103257/

**Comment by softvision-miralobontiu** [comment]
I confirm that the user can "Delete browsing data", on the latest build 1.0.1922 (#11500615), with Google Pixel (Android 9), Samsung Galaxy S8 (Android 9), Nokia 6 (Android 7.1.1).

@bbinto after tapping "Allow", the progress bar doesn't stop moving like it is still in progress. But, if you go back to Homescreen the data was deleted. The user doesn't know if the deletion is complete or not. Please see the attached GIF.

![2019_05_30](https://user-images.githubusercontent.com/40694158/58630681-d8a18b80-82e8-11e9-822b-cab5ba3953c1.gif)

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest build 1.0.1922 (#11530616), with Huawei P9 Lite (Android 6), Google Pixel (Android 9), and Nokia 6 (Android 7.1.1).
